plugins/playstation/src/playstation-uri-game-factory.vala

```diff
@@ -0,0 +1,125 @@
+// This file is part of GNOME Games. License: GPLv3
+
+public class Games.PlayStationUriGameFactory : Object, UriGameFactory {
+	public delegate Game GameForUri (string uri) throws Error;
+
+	private const uint HANDLED_URIS_PER_CYCLE = 5;
+
+	private GameForUri game_for_uri;
+	private UriTest? uri_validity_test;
+	private string[] uris;
+	private Gtk.Image media_image;
+	private GameinfoDoc gameinfo;
+
+	private HashTable<string, string> discs;
+	private HashTable<string, GenericSet<string> > disc_sets;
+	private HashTable<string, Game> games;
+
+	public PlayStationUriGameFactory (GameForUri game_for_uri, Gtk.Image media_image, GameinfoDoc gameinfo, UriTest? uri_validity_test = null) {
+		this.game_for_uri = game_for_uri;
+		this.uri_validity_test = uri_validity_test;
+		this.uris = {};
+		this.media_image = media_image;
```

Comment:
There is probably no need to get the image from the constructor: similarly to `game_for_uri` we know the kind of games we are building here, hence we can handle this ourselves inside the class.